### PR TITLE
Readonly mode added. - Version upgraded to 3.5.4

### PR DESCRIPTION
**Pull request by mkapiczy**
Readonly mode added in only to disable code edition.
Style readonly added and removed dynamically with property. Style can be specified from outside of the codemirror (cuesta).

### CODE REVIEW

File: codemirror.vue [hunk 1]
```diff
@@ -188,6 +192,15 @@ export default {
         this.editor.scrollTo(scrollInfo.left, scrollInfo.top)
       }
     },
+    readonly () {
+      this.editor.setOption("readOnly", this.readonly)
+      if (this.readonly) {
+        document.getElementsByClassName('CodeMirror')[0].className += ' readonly'
```
Comment (jonathanbp) [inline codemirror.vue]: Why not use `this.$el` here, @mkapiczy ?

File: codemirror.vue [hunk 2]
```diff
@@ -188,6 +192,15 @@ export default {
...
+      } else {
+        let codeMirror = document.getElementsByClassName('CodeMirror')[0]
```
Comment (jonathanbp) [inline codemirror.vue]: ... and here

File: codemirror.vue (hunk 1 above)
Comment (mkapiczy) [reply codemirror.vue]: this.$el returns a textarea instance which is behing the Codemirror element.

Comment (jonathanbp) [reply codemirror.vue]: Then `document.querySelector('.CodeMirror', this.$el)` or possibly `document.querySelector('.CodeMirror', this.$el.parentElement)`.

File: codemirror.vue [hunk 3]
```diff
@@ -194,11 +194,11 @@ export default {
     },
     readonly () {
       this.editor.setOption("readOnly", this.readonly)
+      let codeMirrorEl = document.querySelector('.CodeMirror-wrap', this.$el.parentElement)
       if (this.readonly) {
-        document.getElementsByClassName('CodeMirror-wrap')[0].className += ' readonly'
+        codeMirrorEl.className += ' readonly'
```
Comment (jonathanbp) [inline codemirror.vue]: Use `codeMirrorEl.classList.add`.

File: codemirror.vue [hunk 4]
```diff
@@ -194,11 +194,11 @@ export default {
...
       } else {
-        let codeMirror = document.getElementsByClassName('CodeMirror-wrap')[0]
-        codeMirror.className = codeMirror.className.replace('readonly', '')
+        codeMirrorEl.className = codeMirrorEl.className.replace('readonly', '')
```
Comment (jonathanbp) [inline codemirror.vue]: `codeMirrorEl.classList.remove` ;)

**Comment by jonathanbp** [comment]
Published, @mkapiczy:

```bash
~/Development/go/src/gitlab.com/sirenia/vue2-codemirror-lite-js master* 24s
❯ npm publish
+ vue2-codemirror-lite-js@3.5.4
```

**Pull request merged**